### PR TITLE
Special quoting logic for NULLS & ""

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: go/cmd/dolt/commands/sql.go [hunk 1]
```diff
@@ -1130,10 +1130,6 @@ func (se *sqlEngine) prettyPrintResults(ctx context.Context, sqlSch sql.Schema,
 
 	// Parts of the pipeline depend on the output format, such as how we print null values and whether we pad strings.
 	switch se.resultFormat {
-	case formatCsv:
```
Comment (zachmu) [inline go/cmd/dolt/commands/sql.go:1133]: Was this just unnecessary? Already not printing nil values?

File: go/libraries/doltcore/table/untyped/csv/writer.go [hunk 2]
```diff
@@ -30,14 +34,15 @@ import (
 
 // WriteBufSize is the size of the buffer used when writing a csv file.  It is set at the package level and all
```
Comment (zachmu) [inline go/libraries/doltcore/table/untyped/csv/writer.go:35]: Update comment with new name

File: go/libraries/doltcore/table/untyped/csv/writer.go [hunk 3]
```diff
@@ -97,42 +105,150 @@ func (csvw *CSVWriter) GetSchema() schema.Schema {
...
+		csvw.wr = nil
 		return errCl
 	} else {
 		return errors.New("Already closed.")
 	}
 }
+
+// write is directly copied from csv.Writer.Write() with the addition of the `isNull []bool` parameter
+// this method has been adapted for Dolt's special quoting logic, ie `10,,""` -> (10,NULL,"")
+func (csvw *CSVWriter) write(record []string, isNull []bool) error {
```
Comment (zachmu) [inline go/libraries/doltcore/table/untyped/csv/writer.go:154]: Small nit: just make this take []*string and pass nils for the nil strings. No need for the second array.

File: go/cmd/dolt/commands/sql.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/cmd/dolt/commands/sql.go:1133]: `NULL`/`nil` printing is handled by the sink now. We need to eliminate this transform to be able to disambiguate between `NULL` values and empty strings

**Pull request merged**